src/components/editor/ArticleListController.jsx

```diff
@@ -0,0 +1,109 @@
+      let page = this.props.params.page;
+      let maxPage = data ? Math.floor(data.total/pageLength) + (data.total%pageLength > 0) : null;
+      let firstIndex = pageLength*(page-1);
+
+      // If trying to access inacessible page, redirect to page 1
+      if (page < 1 || page > maxPage) {
+        return <p>You have tried accessing a page that doesn't exist. Please press <a href="/articles/page/1/">this link</a> to return to page 1.
+          If you believe this was unintended and there is an error with the website please contact the web development team of The Gazelle.</p>;
+      }
+
+      let indexConverter = data ? [...Array(Math.min(pageLength, data.total-firstIndex-5)).keys()].map((val) => {
+        return val+firstIndex
+      }) : null;
+      let pathStart = "/articles/page/" + page + "/";
+
+    return (
+      <div style={{height: "100%", padding: 0, margin: 0}}>
+        <div style={{height: "100%", width: "50%", float: "left"}}>
+          <h3>Articles</h3>
+          <p style={{width: "75%"}}>Here the 50 most recent articles are shown in the list, if you have an older article in mind,
+          you can find the slug of it either in the Ghost editor, or in URL when you access the website on thegazelle.org
+          and input it below</p>
+          <form onSubmit={this.handleSubmit}>
+            <input type="text" value={this.state.slugSearchValue} placeholder="Input Slug" onChange={this.handleUpdateSlugValue} />
+            <input type="submit" value="Submit" />
+          </form>
+          <br/>
+          {!this.state.ready ? 
+            <p>loading...</p> :
+            <List 
+              data={data}
+              indexConverter={indexConverter}
+              maxHeight="100%"
+              maxWidth="75%"
+              style={{width: "100%", height: "50%", float: "left"}}
+              link={pathStart+"article/slug/"}
+            />
+          }
+          {!this.state.ready ?
+            null :
+            <div style={{width: "75%"}}>
+              <div style={{width: "33%", float: "left"}}>
+                <button type="button" style={{float: "left"}} onClick={this.handlePageChange.bind(this, -1)} disabled={page > 1 ? "" : "disabled"}>Previous Page</button>
+              </div>
+              <div style={{width: "34%", float: "left"}}>
+                <div style={{height: "5px"}} />
+                <center style={{fontFamily: "'Times New Roman', Times, serif", fontSize: "120%"}}>Page {page} of {maxPage}</center>
+              </div>
+              <div style={{width: "33%", float: "left"}}>
+                <button type="button" style={{float: "right"}} onClick={this.handlePageChange.bind(this, 1)} disabled={page < maxPage ? "" : "disabled"}>Next Page</button>
+              </div>
+            </div>
+          }
+        </div>
+        <div style={{width: "50%", height: "100%", float: "left"}}>
+          {this.props.children}
```

Comment:
Rather I meant, shouldn't both sides re render, or at least render with new
properties. But okay whichever works for you it's fine actually

On Sat, Jul 16, 2016, 1:47 AM Emil Goldsmith Olesen <
notifications@github.com> wrote:

> In src/components/editor/ArticleListController.jsx
> <https://github.com/thegazelle-ad/gazelle-front-end/pull/9#discussion_r71050677>
> :
>
> > +            <div style={{width: "75%"}}>
> > +              <div style={{width: "33%", float: "left"}}>
> > +                <button type="button" style={{float: "left"}} onClick={this.handlePageChange.bind(this, -1)} disabled={page > 1 ? "" : "disabled"}>Previous Page</button>
> > +              </div>
> > +              <div style={{width: "34%", float: "left"}}>
> > +                <div style={{height: "5px"}} />
> > +                <center style={{fontFamily: "'Times New Roman', Times, serif", fontSize: "120%"}}>Page {page} of {maxPage}</center>
> > +              </div>
> > +              <div style={{width: "33%", float: "left"}}>
> > +                <button type="button" style={{float: "right"}} onClick={this.handlePageChange.bind(this, 1)} disabled={page < maxPage ? "" : "disabled"}>Next Page</button>
> > +              </div>
> > +            </div>
> > +          }
> > +        </div>
> > +        <div style={{width: "50%", height: "100%", float: "left"}}>
> > +          {this.props.children}
>
> I've been thinking of restructuring so that the right hand side stays
> there even when you change to /authors as I feel like it would be nice but
> not necessary. But for now I'm at least just getting it to work like this
> as it's what I'm doing now and I don't think the other functionality is
> that useful, although elegant. But yeah, for now if you change to /authors
> the right hand side will dissappear. And in /authors it will be the same
> code structure with it rendering children.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/thegazelle-ad/gazelle-front-end/pull/9/files/cd3df8271a1301dd380cbce86fc65f61794aa5fd#r71050677>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADAFzYc9uAWj7OjEPi2-1IJNYVVox8HMks5qWA3lgaJpZM4JMWcC>
> .
>